ui/src/api/services/destinationService.ts

```diff
@@ -151,4 +151,19 @@ export const destinationService = {
 		)
 		return response.data
 	},
+
+	checkDestinationNameUnique: async (
+		destinationName: string,
+	): Promise<{ unique: boolean }> => {
+		try {
+			const response = await api.post<{ unique: boolean }>(
+				`${API_CONFIG.ENDPOINTS.PROJECT(API_CONFIG.PROJECT_ID)}/check-unique`,
+				{ name: destinationName, entity_type: "destination" },
```

Comment:
```suggestion
				{ name: destinationName, entity_type: ENTITY_TYPES.DESTINATION },
```

Similar to what you did in source and destination service.

Reply:
Done